src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.SourceBuild.Tasks.LeakDetection/CheckForPoison.cs

```diff
@@ -213,6 +231,31 @@ internal IEnumerable<PoisonedFileEntry> GetPoisonedFiles(IEnumerable<string> ini
             return poisons;
         }
 
+        private List<string> GetAllNonShippingPackages()
+        {
+            List<string> nonShippingPackages = new List<string>();
+
+            if (NonShippingPackagesListFiles != null)
+            {
+                foreach (ITaskItem item in NonShippingPackagesListFiles)
+                {
+                    using (StreamReader sr = new StreamReader(item.ItemSpec))
```

Comment:
File.ReadAllLines would simplify this code a bit and would allow the use of linq;

```
if (NonShippingPackagesListFiles != null)
{
    return NonShippingPackagesListFiles
        .SelectMany(item => Files.ReadAllLines(item.ItemSpec))
        .Distinct()
        .ToList();
}
else
{
    Enumerable.Empty();
}
```

Reply:
Thanks - I will incorporate this.